### PR TITLE
Fix #300: `min-required` not evaluated for auto-install

### LINKED ISSUE #300: min-required doesn't work with auto-install
```
$ echo 'min-required' > .terraform-version
$ terraform init
version 'min-required' is not installed (set by [...]/.terraform-version). Installing now as TFENV_AUTO_INSTALL==true
Terraform v1.0.0 is already installed
/opt/tfenv/libexec/tfenv-exec: line 105: /var/lib/tfenv/versions/min-required/terraform: No such file or directory
```

I think there's two problems here with when 'min-required' is evaluated:

1. it's thought (presumably always) not to be installed, and tries to install; at which point in this case its value was already installed;
2. it's again not evaluated in (or in the call to) `tfenv-exec`, so it tries to find the `terraform` binary in a `min-required` dir that will never exist

### PR DESCRIPTION
**Pull request by OJFord**
Fixes #300

### CODE REVIEW

File: libexec/tfenv-version-name [hunk 1]
```diff
@@ -137,4 +137,8 @@ if [[ ! -d "${TFENV_CONFIG_DIR}/versions/${TFENV_VERSION}" ]]; then
   log 'debug' "version '${TFENV_VERSION}' is not installed (set by ${TFENV_VERSION_SOURCE})";
 fi;
 
+if [[ "${TFENV_VERSION}" = min-required ]]; then
```
Comment (Zordrak) [inline libexec/tfenv-version-name:140]: Did you mean:
`if [[ "${TFENV_VERSION}" == "min-required" ]]; then` ?

Comment (OJFord) [reply libexec/tfenv-version-name:140]: It's the same right? No problem if you prefer though.

**Comment by OJFord** [comment]
Now with a test, since you asked on my others (quite rightly) and this was an easy one to  get into them with. 🙂

**Comment by OJFord** [comment]
Just a couple of aesthetic changes as I noticed it wasn't consistent with the style in use.

**Review by Zordrak — APPROVED** [review]

**Pull request merged**